From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect struct mlx5_td
Patch-mainline: never, kabi
References: kabi

In networking-stable-19_04_10, upstream commit
80a2a9026b24c6bd34b8d58256973e22270bedec (net/mlx5e: Add a lock on tir
list) added a list_lock to struct mlx5_td. It made the kABI checker to
complain.

Given the structure is private to mlx5, hide the change from the kABI
checker.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 include/linux/mlx5/driver.h |    2 ++
 1 file changed, 2 insertions(+)

--- a/include/linux/mlx5/driver.h
+++ b/include/linux/mlx5/driver.h
@@ -746,7 +746,9 @@ struct mlx5_pagefault {
 
 struct mlx5_td {
 	/* protects tirs list changes while tirs refresh */
+#ifndef __GENKSYMS__
 	struct mutex     list_lock;
+#endif
 	struct list_head tirs_list;
 	u32              tdn;
 };
